exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServer.java

```diff
@@ -75,7 +76,8 @@ public static PrometheusHttpServerBuilder builder() {
     // we configure prometheus with a single thread executor such that requests are handled
     // sequentially.
     if (memoryMode == MemoryMode.REUSABLE_DATA) {
```

Comment:
```suggestion
    if (memoryMode == MemoryMode.REUSABLE_DATA || executor == null) {
```
We want this to apply also when the use hasn't explicitly set the executor

Reply:
Perhaps we don't need this conditional expression, as the `executor` is null by default, and then the Prometheus http server will create a new multi-thread pool executor internally. If we include the `executor == null` expression, the Prometheus http server will always handle http requests using a single thread.

Reply:
I was under the impression that the default thread pool for prometheus wasn't daemon, but looking closer at the `prometheus-metrics-exporter-httpserver` source code it is in fact a daemon thread pool.

But even if that wasn't the case, this suggestion wasn't the right way to accomplish that since it effectively changes the default to a single thread pool. 

Please disregard!